### PR TITLE
*: enable bodyclose linter

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

ref #4322

### What is changed and how it works?

Enable bodyclose linter.

### Check List

Tests

- Unit test
### Release note

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: server/api/util.go [hunk 1]
```diff
@@ -127,17 +127,18 @@ func patchJSON(client *http.Client, url string, body []byte) error {
...
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
 	if err != nil {
-		return nil, err
+		return http.StatusBadRequest, err
 	}
+	// TODO: find a better way to return status code.
 	res, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return http.StatusInternalServerError, err
```
Comment (disksing) [inline server/api/util.go:138]: can we `return 0, err` here?

Comment (rleungx) [reply server/api/util.go:138]: The caller will get 0 as the status code which is not good. I think an alternative is to return the status code according to the detailed error. But I'm going to leave it for now.

File: tests/server/api/api_test.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ func (s *serverTestSuite) TestReconnect(c *C) {
 		if name != leader {
 			res, e := http.Get(s.GetConfig().AdvertiseClientUrls + "/pd/api/v1/version")
 			c.Assert(e, IsNil)
+			res.Body.Close()
```
Comment (JmPotato) [inline tests/server/api/api_test.go:73]: Why not `defer`?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: tests/server/api/api_test.go (hunk 2 above)
Comment (rleungx) [reply tests/server/api/api_test.go:73]: It's a for loop, closing it directly is better.

File: server/api/util.go (hunk 1 above)
Comment (disksing) [reply server/api/util.go:138]: 0 is ok to use as an _invalid status code_ when err is not nil. If you construct status code in client side, we cannot distiguish the cases that we receive the status code from server or construct it from client.

Comment (rleungx) [reply server/api/util.go]: Make sense, change to 0.

**Comment by rleungx** [comment]
@JmPotato @disksing PTAL

**Review by JmPotato — APPROVED** [review]

**Pull request merged**